api/provider/aws/service_read.go

```diff
@@ -9,17 +9,23 @@ import (
 )
 
 func (s *ServiceProvider) Read(serviceID string) (*models.Service, error) {
-	model := &models.Service{}
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return nil, err
+	}
 
-	clusterName := addLayer0Prefix(s.Config.Instance(), model.EnvironmentID)
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), environmentID)
+	clusterName := fqEnvironmentID
 	service, err := s.readService(clusterName, serviceID)
 	if err != nil {
 		return nil, err
 	}
 
-	model.DesiredCount = aws.Int64Value(service.DesiredCount)
-	model.RunningCount = aws.Int64Value(service.RunningCount)
-	model.PendingCount = aws.Int64Value(service.PendingCount)
+	model := &models.Service{
```

Comment:
you forgot to add service id, environment id, loadbalancer id

Reply:
✅